routing/segment.hpp

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include "routing/road_point.hpp"
+
+#include "std/cstdint.hpp"
+#include "std/sstream.hpp"
+#include "std/string.hpp"
+
+namespace routing
+{
+class Segment final
+{
+public:
+  Segment() = default;
+
+  Segment(uint32_t featureId, uint32_t segmentId, bool forward)
+    : m_featureId(featureId), m_segmentId(segmentId), m_forward(forward)
+  {
+  }
+
+  uint32_t GetFeatureId() const { return m_featureId; }
+  uint32_t GetSegmentId() const { return m_segmentId; }
+  bool IsForward() const { return m_forward; }
+  uint32_t GetPointId(bool front) const
+  {
+    return m_forward == front ? m_segmentId + 1 : m_segmentId;
+  }
+  RoadPoint GetRoadPoint(bool front) const { return {m_featureId, GetPointId(front)}; }
```

Comment:
Probably it's better to call construction of RoadPoint directly? I mean: RoadPoint(m_featureId, GetPointId(front))?